sdk/eidolon_ai_sdk/cpu/conversational_apu.py

```diff
@@ -180,11 +182,23 @@ async def _llm_execution_cycle(
             if stream_collector.get_content():
                 logger.info(f"LLM Response: {stream_collector.get_content()}")
 
-            assistant_message = AssistantMessage(
-                type="assistant",
-                content=stream_collector.get_content() or "",
-                tool_calls=[tce.tool_call for tce in tool_call_events],
-            )
+            if isinstance(self.llm_unit, ToolCallLLMWrapper) and tool_call_events:
+                content = stream_collector.get_content() or ""
+                content += "\nCall the following tools\n"
+                for tool_call in tool_call_events:
+                    content += f"\n{tool_call.tool_call.model_dump_json()}"
+                assistant_message = AssistantMessage(
+                    type="assistant",
+                    content=content,
+                    tool_calls=[],
```

Comment:
Is this tool_calls variable used anywhere else? ... Like, is it relied upon downstream anywhere to contain something?

Reply:
Yes, it will affect the ui, so it will need to be translated on the way out. This should be transparent to every layer above the llm unit